tracing/http_test.go

```diff
@@ -94,3 +95,77 @@ func TestHTTPSpanMiddleware(t *testing.T) {
 		t.Errorf("spans[0].TraceID = %s, want %s", got, traceID)
 	}
 }
+
+func TestHTTPSpanMiddlewareIgnoringPaths(t *testing.T) {
+	cfg := config.Config{
+		Backend: config.Zipkin,
+		Debug:   true,
+	}
+
+	// Create tracer with reporter recorder
+	reporter, co := FakeZipkinExporter()
+	defer reporter.Close()
+	oct := NewOpenCensusTracer(co)
+	defer oct.Finish()
+
+	if err := oct.ApplyConfig(&cfg); err != nil {
+		t.Errorf("Failed to apply tracer config: %v", err)
+	}
+
+	paths := []string{"/readyz"}
+	middleware := HTTPSpanMiddlewareIgnoringPaths(paths...)(&testHandler{})
+
+	testCases := map[string]struct {
+		path   string
+		traced bool
+	}{
+		"traced": {
+			path:   "/",
+			traced: true,
+		},
+		"ignored": {
+			path:   paths[0],
+			traced: false,
+		},
+	}
+	for n, tc := range testCases {
+		t.Run(n, func(t *testing.T) {
+			var lastWrite []byte
+			fw := fakeWriter{lastWrite: &lastWrite}
+
+			url := url.URL{
+				Scheme: "http",
+				Host:   "test.example.com",
+				Path:   tc.path,
+			}
+			req, err := http.NewRequest("GET", url.String(), nil)
+			if err != nil {
+				t.Errorf("Failed to make fake request: %v", err)
+			}
+			traceID := "821e0d50d931235a5ba3fa42eddddd8f"
+			req.Header["X-B3-Traceid"] = []string{traceID}
```

Comment:
Prefer:

```golang
req.Header.Set("X-B3-Traceid", traceId)
req.Header.Set("X-B3-Spanid", "....")
```

Set will create the slice and so forth for you.

Reply:
Done.